### ISSUE TITLE
New Classification Metric - average_precision_score

### ISSUE THREAD

**Issue by mail4umar** [vertica/VerticaPy#663]
Average precision score is another metric for classification. It is slightly different than precision-recall curve area under the curve. PRC AUC assumes a kind linear relationship. But average precision score does not have that. 

Scikit learn also has a function to compute this score: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html
